modules/ocf/manifests/staff_users.pp

```diff
@@ -1,11 +1,10 @@
 # Create home directories for staff members if NFS isn't mounted.
 
-class ocf::staff_users {
+class ocf::staff_users($noop = false) {
   # Only create home directories if:
   #   - we don't mount NFS
-  #   - we're a server (not a desktop)
-  #   - we're not the file server
-  if !str2bool($::ocf_nfs) and $::type == 'server' and !tagged('ocf_filehost') {
+  #   - it hasn't been turned off in Hiera
+  if !str2bool($::ocf_nfs) and !$noop {
```

Comment:
This does what's wanted, but it just seems more fragile to me, since it's essentially defined in two different places which systems don't create staff directories (facts and hiera). In my opinion it was nicer when it was restricted to servers only, since now the noop must explicitly opt-out, which seems like an issue waiting to happen. Maybe it could be made opt-in instead and then enabled for all servers? Not sure that's much better since then jaws is still an exception, but this current system seems like it could sometimes have problems otherwise.